core/src/main/java/io/questdb/cutlass/text/types/IPv4Adapter.java

```diff
@@ -66,7 +75,36 @@ public void write(TableWriter.Row row, int column, DirectUtf8Sequence value) thr
         row.putIPv4(column, SqlKeywords.isNullKeyword(value) ? Numbers.IPv4_NULL : parseIPv4(value));
     }
 
+    private static boolean hasDots(DirectUtf8Sequence value) {
+        final int len = value.size();
+        int i = 0;
+        while (i < len) {
+            if (i < len - 7) {
```

Comment:
this logic is buggy, it should be `len-8` and generally this code should be more efficient.

- this method will fail for a string at exactly length 8
- it should probably be two separate loops to make it easier to read
- the method should probably move somewhere to Utf8s class and tested for edge cases